rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -212,7 +212,15 @@ bool Player::play()
       "Invalid delay value: " << play_options_.delay.nanoseconds() << ". Delay is disabled.");
   }
 
-  RCLCPP_INFO_STREAM(get_logger(), "Playback duration value: " << play_until_time_);
+  rcutils_time_point_value_t play_until_timestamp = -1;
+  if (play_options_.playback_duration >= rclcpp::Duration(0, 0) ||
+    play_options_.playback_until_timestamp >= rcutils_time_point_value_t{0})
+  {
+    play_until_timestamp = std::max(
+      starting_time_ + play_options_.playback_duration.nanoseconds(),
+      play_options_.playback_until_timestamp);
+  }
+  RCLCPP_INFO_STREAM(get_logger(), "Playback until timestamp: " << play_until_timestamp);
```

Comment:
@agalbachicar We have moved logic for determining actual `play_until_timestamp` in constructor and made it as private member.  
Please see comment https://github.com/ros2/rosbag2/pull/960#issuecomment-1144020905  and relevant changes in https://github.com/ros2/rosbag2/pull/960/commits/38ed671366f55ab31208c65dc70ca0c2cc17c0ac . Also we've got rid from `play_until_timestamp` parameter in `play_messages_from_queue(play_until_timestamp);` method since it currently using private member variable.
Please follow to the aforementioned changes and adjust your code accordingly.